source/common/common/utility.cc

```diff
@@ -332,4 +332,27 @@ std::regex RegexUtil::parseRegex(const std::string& regex, std::regex::flag_type
   }
 }
 
+// https://en.wikipedia.org/wiki/Algorithms_for_calculating_variance#Online_algorithm
+void WelfordStandardDeviation::update(double newValue) {
+  ++count_;
+  const double delta = newValue - mean_;
+  mean_ += delta / count_;
+  const double delta2 = newValue - mean_;
+  m2_ += delta * delta2;
+}
+
+double WelfordStandardDeviation::computeVariance() const {
+  if (count_ < 2) {
+    return std::nan("");
+  }
+  return m2_ / (count_ - 1);
+}
+
+double WelfordStandardDeviation::computeStandardDeviation() const {
+  const double variance = computeVariance();
+  // It seems very difficult for variance to go negative, but from the calculation in update()
+  // above, I can't quite convince myself it's impossible, so put in a guard to be sure.
```

Comment:
Should this just be an `ASSERT`, since it shouldn't be possible to have negative variance..

Reply:
I agree it shouldn't be possible, but this algorithm is an approximation and it's not super-obvious to me that it can't happen given the way it's calculated.  So I thought it's easier just to guard against it and avoid runtime exceptions.